climada/entity/exposures/litpop/gpw_population.py

```diff
@@ -0,0 +1,164 @@
+        target year for data extraction
+    gpw_version : int (optional)
+        Version number of GPW population data.
+        The default is 11
+    data_dir : Path (optional)
+        Path to directory with GPW data.
+        The default is SYSTEM_DIR.
+    layer : int (optional)
+        relevant data layer in input TIFF file to return.
+        The default is 0 and should not be changed without understanding the
+        different data layers in the given TIFF file.
+    verbatim : bool (optional):
+        if False, output in LOGGER is surpressed. Default is True.
+
+    Returns
+    -------
+    pop_data : 2D numpy array
+        contains extracted population count data per grid point in shape
+        first dimension is lat, second dimension is lon.
+    meta : dict
+        contains meta data per array, including "transform" with 
+        meta data on coordinates.
+    global_transform : Affine instance
+        contains six numbers, providing transform info for global GWP grid.
+        global_transform is required for resampling on a globally consistent grid
+    """
+    # check whether GPW input file exists and get file path
+    file_path = get_gpw_file_path(gpw_version, reference_year, data_dir=data_dir,
+                                  verbatim=verbatim)
+
+    # open TIFF and extract cropped data from input file:
+    src = rasterio.open(file_path)
+    global_transform = src.transform
+    pop_data, out_transform = rasterio.mask.mask(src, [geometry], crop=True,
+                                                  nodata=0)
+
+    # extract and update meta data for cropped data and close src:
+    meta = src.meta
+    meta.update({"driver": "GTiff",
+                 "height": pop_data.shape[1],
+                 "width": pop_data.shape[2],
+                 "transform": out_transform})
+    src.close()
+    return pop_data[layer,:,:], meta, global_transform
+
+def get_gpw_file_path(gpw_version, reference_year, data_dir=None, verbatim=True):
+    """Check available GPW population data versions and year closest to
+    reference_year and return full path to TIFF file.
+
+    Parameters
+    ----------
+    gpw_version : int (optional)
+        Version number of population data.
+        The default is 11
+    reference_year : int (optional)
+        Data year is selected as close to reference_year as possible.
+        The default is 2020.
+    data_dir : pathlib.Path (optional)
+        Absolute path where files are stored. Default: SYSTEM_DIR
+
+    Raises
+    ------
+    FileExistsError
+
+    Returns
+    -------
+    pathlib.Path : path to input file with population data
+    """
+    if gpw_version is None:
+        gpw_version = []
+    elif isinstance(gpw_version, int):
+        gpw_version = [gpw_version]
+
+    if data_dir is None:
+        data_dir = SYSTEM_DIR
+    # find closest year to reference_year with data available:
+    year = YEARS_AVAILABLE[np.abs(YEARS_AVAILABLE - reference_year).argmin()]
```

Comment:
If the year is exactly between available years, the older one is used. Maybe invert the YEARS_AVAILABLE to then select the most recent one?